austrakka/components/metadata/__init__.py

```diff
@@ -28,33 +28,61 @@ def metadata(ctx):
 @click.argument('file', type=click.File('rb'))
 @opt_proforma()
 @opt_blanks_delete()
-def submission_add(file: BufferedReader, proforma: str, blanks_will_delete: bool = False):
+@opt_batch_size(help='The number of rows to split the metadata upload into before uploading. '
+                     'If the file size is below this value, the file will not be split. '
+                     'An upload record will be recorded in the database per batch, and '
+                     'validation and success messages will be returned per batch. '
+                     'Note that if an Excel file is batched, the only the submission '
+                     'worksheet will be uploaded. '
+                     'A negative or 0 value can be used to indicate no batching.',
+                default=5000)
+def submission_add(
+        file: BufferedReader, 
+        proforma: str, 
+        blanks_will_delete: bool,
+        batch_size: int):
     """Upload metadata submission to AusTrakka"""
-    add_metadata(file, proforma, blanks_will_delete)
+    add_metadata(file, proforma, blanks_will_delete, batch_size)
 
 
 @metadata.command('append')
 @click.argument('file', type=click.File('rb'))
 @opt_proforma()
 @opt_blanks_delete()
-def submission_append(file: BufferedReader, proforma: str, blanks_will_delete: bool = False):
+@opt_batch_size(help='The number of rows to split the metadata upload into before uploading. '
+                     'If the file size is below this value, the file will not be split. '
+                     'An upload record will be recorded in the database per batch, and '
+                     'validation and success messages will be returned per batch. '
```

Comment:
Lets define the help text in one place at the top of the file, and reuse for add/append.